### ISSUE TITLE
Bluetooth stopped working on a thinkpad with 22.04 after kernel upgrade to 6.0.x (Bluetooth: hci0: unexpected cc 0x2060 length: 1 < 7, Bluetooth: hci0: Opcode 0x2060 failed: -38, Bluetooth: hci0: command tx timeout)

### ISSUE THREAD

**Issue by bicatu** [pop-os/pop#2687]
**Distribution (run `cat /etc/os-release`):**
 
 NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os

 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

**Issue/Bug Description:**

Bluetooth does not work in any kernel 6.0.x, only with the 5.16

**Steps to reproduce (if you know):**

Install the distro 22.04 in a Thinkpad T14 2022. The Bluetooth is recognized and can be used.
Perform an upgrade (apt update/upgrade), reboot to the kernel 6.0.x and the Bluetooth appears disabled and can't be enabled.

**Expected behavior:**

the Bluetooth device to appear

**Other Notes:**

Already followed https://support.system76.com/articles/bluetooth/ and the fact that if I reboot the system back to the 5.16. kernel it works is odd.

When booted under 6.0.x kernel
- lsmod | grep Bluetooth

```
bluetooth             880640  15 btrtl,btmtk,btintel,btbcm,bnep,btusb
ecdh_generic           16384  1 Bluetooth
```

- sudo dmesg | grep Bluetooth
```
[   11.645116] Bluetooth: Core ver 2.22
[   11.645160] Bluetooth: HCI device and connection manager initialized
[   11.645167] Bluetooth: HCI socket layer initialized
[   11.645170] Bluetooth: L2CAP socket layer initialized
[   11.645177] Bluetooth: SCO socket layer initialized
[   11.680846] Bluetooth: hci0: HCI Enhanced Setup Synchronous Connection command is advertised, but not supported.
[   11.772840] Bluetooth: hci0: unexpected cc 0x2060 length: 1 < 7
[   11.772857] Bluetooth: hci0: Opcode 0x2060 failed: -38
[   11.826991] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[   11.826996] Bluetooth: BNEP filters: protocol multicast
[   11.827001] Bluetooth: BNEP socket layer initialized
[   13.780965] Bluetooth: hci0: command tx timeout
```
- sudo systemctl status Bluetooth

```
● bluetooth.service - Bluetooth service
     Loaded: loaded (/lib/systemd/system/bluetooth.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2022-11-07 20:06:38 EST; 4min 30s ago
       Docs: man:bluetoothd(8)
   Main PID: 5709 (bluetoothd)
     Status: "Running"
      Tasks: 1 (limit: 52616)
     Memory: 632.0K
        CPU: 35ms
     CGroup: /system.slice/bluetooth.service
             └─5709 /usr/lib/bluetooth/bluetoothd

Nov 07 20:06:38 pop-os systemd[1]: Starting Bluetooth service...
Nov 07 20:06:38 pop-os bluetoothd[5709]: Bluetooth daemon 5.64
Nov 07 20:06:38 pop-os systemd[1]: Started Bluetooth service.
Nov 07 20:06:38 pop-os bluetoothd[5709]: Starting SDP server
Nov 07 20:06:38 pop-os bluetoothd[5709]: Bluetooth management interface 1.22 initialized
```

**Comment by bicatu** [comment]
Nothing? Am I the only one with this issue?

**Comment by bicatu** [comment]
Hi @heydemo in my case, the Bluetooth icon does not even appear in the UI. It is as if I do not have this capability enabled on the laptop.
What is bizarre is that I have a friend with the same model of the laptop and for him it works without an issue.

I have been testing with all versions of the linux kernel 6 that popOs is releasing without luck.

**Comment by bicatu** [comment]
When booting under 5.16 kernel -- where everything works -- I get

```
sudo dmesg | grep Bluetooth
[    4.007775] Bluetooth: Core ver 2.22
[    4.007934] Bluetooth: HCI device and connection manager initialized
[    4.007942] Bluetooth: HCI socket layer initialized
[    4.007945] Bluetooth: L2CAP socket layer initialized
[    4.007949] Bluetooth: SCO socket layer initialized
[    4.273112] Bluetooth: hci0: Failed to read codec capabilities (-56)
[    4.276056] Bluetooth: hci0: Failed to read codec capabilities (-56)
[    4.279061] Bluetooth: hci0: Failed to read codec capabilities (-56)
[    4.282039] Bluetooth: hci0: Failed to read codec capabilities (-56)
[    4.283296] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[    4.283299] Bluetooth: BNEP filters: protocol multicast
[    4.283306] Bluetooth: BNEP socket layer initialized
[    4.285286] Bluetooth: hci0: Failed to read codec capabilities (-56)
[    4.969377] Bluetooth: RFCOMM TTY layer initialized
[    4.969384] Bluetooth: RFCOMM socket layer initialized
[    4.969389] Bluetooth: RFCOMM ver 1.11
```

**Comment by bicatu** [comment]
I found a recent thread on a Fedora list that points to a kernel bug.

https://bugzilla.redhat.com/show_bug.cgi?id=2149136#c17

Most likely my colleague with the same model uses a different chipset that is immune. According to the the post it has been fixed in the kernel source code. Now it is a matter of waiting to see how long does it take to make its way to popOs.

**Comment by Comarco** [comment]
I have a HP Zbook G15. Bluetooth stopped working as soon as I updated to kernel v6

`uname -a
Linux pop-os 6.0.12-76060006-generic #202212290932~1673966427~22.04~e15d5b5 SMP PREEMPT_DYNAMIC Tue J x86_64 x86_64 x86_64 GNU/Linux`

In the UI, I have the Bluetooth setting but it has the message "No Bluetooth Found. Plug in a dongle to use bluetooth"

`cat /etc/os-release
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os`

I have no idea what module is managing the bluetooth device on this machine.

Could it be:
`Network controller [0280]: Intel Corporation Wireless 8260 [8086:24f3] (rev 3a)
	Subsystem: Intel Corporation Dual Band Wireless-AC 8260 [8086:1010]`

`hcitool scan
Device is not available: No such device`

`rfkill list all
0: phy0: Wireless LAN
	Soft blocked: no
	Hard blocked: no
`

Any help would be appreciated.
(I hope I don't hijack this issue with my own problem, but it looks pretty similar)

**Comment by bicatu** [comment]
This was solved with the latest 6.1 and 6.2 kernels from popOs.

**Comment by scttnlsn** [comment]
I'm having this same issue after upgrading my kernel and I'm currently on `6.2.6`:

```
Linux pop-os 6.2.6-76060206-generic #202303130630~1685473338~22.04~995127e SMP PREEMPT_DYNAMIC Tue M x86_64 x86_64 x86_64 GNU/Linux
```

Is there any config I need to change to get BT working on 6.2?

**Comment by csdvrx** [comment]
> This was solved with the latest 6.1 and 6.2 kernels from popOs.

@bitcatu could you mention which kernel version so I can check the commit that fixed it?
